pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -341,6 +341,9 @@ func validatePipelineParameterVariables(tasks []PipelineTask, params []ParamSpec
 			}
 		}
 
+		if _, ok := parameterNames[p.Name]; ok {
```

Comment:
is it possible to add the same condition for `arrayParameterNames`?

Also, we are still allowing params of different types having same `p.Name`, since we are updating release notes, can we add the check and not permit duplicate param names having different types?

Reply:
This logic catches that issue too as far as I can see since this line does execute for all params regardless of their type. I added more tests to show the effect of it and the fact that it works in that case as well. Let me know if I understood what you said correctly or if you wanted me to do something else.

Reply:
yeah I see, `parameterNames` holds all parameters whereas `arrayParameterNames` is a subset of `parameterNames` and only holds array params. Sorry for confusion 🙏